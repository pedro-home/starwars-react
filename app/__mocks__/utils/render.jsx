import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureStore from 'store';
import Refs from 'contexts/Refs';
import { DEFAULT_REFS } from 'contexts/Refs/constants';

export function renderWithRedux(component, initialState = {}) {
  const store = configureStore(initialState);
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
}

export function renderWithApp(component, initialState = {}, refsValue = DEFAULT_REFS) {
  return renderWithRedux(<Refs.Provider value={refsValue}>{component}</Refs.Provider>, initialState);
}
