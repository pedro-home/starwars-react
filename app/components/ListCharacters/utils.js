import femaleLogo from './assets/female.jpg';
import maleLogo from './assets/male.jpg';
import hermaphroditeLogo from './assets/hermaphrodite.jpg';
import droidLogo from './assets/droid.jpg';
import { OFFSET_SCROLL } from './constants';

export const getCharacterImage = name => {
  let logo;
  switch (name) {
    case 'female':
      logo = femaleLogo;
      break;
    case 'male':
      logo = maleLogo;
      break;
    case 'hermaphrodite':
      logo = hermaphroditeLogo;
      break;
    default:
      logo = droidLogo;
  }

  return logo;
};

export const hasEndedScroll = (offset = OFFSET_SCROLL) => {
  return window.scrollY + window.innerHeight >= document.body.scrollHeight - offset;
};
