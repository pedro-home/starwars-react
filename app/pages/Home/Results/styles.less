.results {
  height: 100vh;

  .center-container {
    display: inline-block;
  }
}
