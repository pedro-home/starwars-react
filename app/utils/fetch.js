function fetch(url, options) {
  return window.fetch(url, options).then(response => {
    if (response.ok) {
      let body;
      const contentType = response.headers.get('content-type');
      if (contentType && contentType.indexOf('application/json') !== -1) {
        body = response.json();
      } else {
        body = response.text();
      }

      return body;
    }

    throw new Error(`${response.status}: ${response.statusText}`);
  });
}

export default fetch;
