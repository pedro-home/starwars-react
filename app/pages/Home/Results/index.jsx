import React, { useContext } from 'react';
import Refs from 'contexts/Refs';
import ListCharacters from 'containers/ListCharacters';

const Section = () => {
  const { setResults } = useContext(Refs);
  return (
    <section
      className="section results"
      ref={elem => {
        setResults(elem);
      }}
    >
      <ListCharacters />
    </section>
  );
};

export default Section;
