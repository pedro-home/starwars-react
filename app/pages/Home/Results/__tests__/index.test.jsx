import React from 'react';
import { renderWithApp } from '__mocks__/utils/render';

import Results from '../index';

describe('Given <Results />', () => {
  let component;

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = renderWithApp(<Results />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render a quote', () => {
      expect(component.querySelector('.quote')).toBeTruthy();
    });
  });
});
