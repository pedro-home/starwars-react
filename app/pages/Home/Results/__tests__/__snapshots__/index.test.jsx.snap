// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Results /> When initializes Then should match the snapshot 1`] = `
<div>
  <section
    class="section results"
  >
    <div
      class="container-center"
    >
      <h4
        class="quote"
      >
        <q>
          Search your feelings, you know you to be true
        </q>
      </h4>
    </div>
  </section>
</div>
`;
